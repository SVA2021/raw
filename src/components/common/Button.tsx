import classNames from 'classnames';
import React, { FC } from 'react';

import styles from './Button.module.scss';

type TButtonProps = {
	className?: string;
	onClick?: React.MouseEventHandler<HTMLButtonElement> | undefined;
	withOffset?: boolean;
	active?: boolean;
};

const Button: FC<TButtonProps> = ({ className = '', withOffset, active, children, ...props }) => (
	<button
		className={classNames(styles.button, {
			[styles.withOffset]: withOffset,
			[styles.active]: active,
		}, className)}
		{...props}
	>
		{children}
	</button>
);

export default Button;
