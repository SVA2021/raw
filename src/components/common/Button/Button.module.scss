@import '../../../styles/variables';
@import '../../../styles/mixins';

.button {
	@include buttonBase;
	@include buttonOffset;

	&:hover {
		box-shadow: 5px 5px 50px $orange, -5px -5px 50px $orange;
		transform: scale(110%);
		transition: transform 0.25s linear;
	}
}

.active {
	@extend .button;
	@include buttonActive;

	&:hover {
		box-shadow: 5px 5px 50px $green, -5px -5px 50px $green;
	}
}

.withOffset {
	margin-left: $offset-basic;
	margin-right: $offset-basic;
}