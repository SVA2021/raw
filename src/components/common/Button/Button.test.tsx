import React from 'react';
import { render, screen } from '@testing-library/react';
import Button from './Button';
import userEvent from '@testing-library/user-event';

describe('Button rendering', () => {

	const handleClicker = jest.fn();

	const setUp = () => render(
		<Button data-testid='mybtn' onClick={handleClicker}>learn</Button>
	);

	it('snapshot test', () => {
		setUp();
		expect(screen.getByTestId('mybtn')).toMatchSnapshot();
	});

	it('find button in document by role', () => {
		setUp();
		expect(screen.getByRole('button')).toBeInTheDocument();
	});

	it('not find button by text', () => {
		setUp();
		expect(screen.queryByText(/another/i)).toBeNull();
	});

	it('find button by text', () => {
		setUp();
		expect(screen.getByText(/learn/i)).toBeInTheDocument();
	});

	it('click button', () => {
		setUp();
		userEvent.click(screen.getByTestId('mybtn'));
		expect(handleClicker).toHaveBeenCalledTimes(1);
	});

	screen.debug();
});
