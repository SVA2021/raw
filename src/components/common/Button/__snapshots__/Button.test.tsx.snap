// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button rendering snapshot test 1`] = `
<button
  class="button"
  data-testid="mybtn"
>
  learn
</button>
`;
